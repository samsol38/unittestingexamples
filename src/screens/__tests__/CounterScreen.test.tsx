import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import CounterScreen from '../counter';

describe('CounterScreen render with snapshot', () => {
  test('matches snapshot', () => {
    const tree = render(<CounterScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('CounterScreen events', () => {
  test('renders initial counter value', () => {
    const {getByText} = render(<CounterScreen />);
    const counterText = getByText('Counter: 0');
    expect(counterText).toBeDefined();
  });

  test('increments counter value when increment button is pressed', () => {
    const {getByText} = render(<CounterScreen />);
    const incrementButton = getByText('Increment');
    const counterText = getByText('Counter: 0');
    fireEvent.press(incrementButton);
    expect(counterText.props.children).toBe('Counter: 1');
  });

  test('decrements counter value when decrement button is pressed', () => {
    const {getByText} = render(<CounterScreen />);
    const decrementButton = getByText('Decrement');
    const counterText = getByText('Counter: 0');
    fireEvent.press(decrementButton);
    expect(counterText.props.children).toBe('Counter: -1');
  });
});
