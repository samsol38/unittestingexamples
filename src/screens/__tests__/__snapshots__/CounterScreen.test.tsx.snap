// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterScreen render with snapshot matches snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      {
        "fontSize": 24,
        "marginBottom": 20,
      }
    }
  >
    Counter: 0
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "blue",
        "borderRadius": 5,
        "marginBottom": 10,
        "opacity": 1,
        "padding": 10,
      }
    }
  >
    <Text
      style={
        {
          "color": "white",
          "fontSize": 16,
        }
      }
    >
      Increment
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "blue",
        "borderRadius": 5,
        "marginBottom": 10,
        "opacity": 1,
        "padding": 10,
      }
    }
  >
    <Text
      style={
        {
          "color": "white",
          "fontSize": 16,
        }
      }
    >
      Decrement
    </Text>
  </View>
</View>
`;
